### PR TITLE
[nghttp3] Add new Port

### PR DESCRIPTION
**Pull request by alagoutte**
**Describe the pull request**

- #### What does your PR fix?
Add nghttp3 Port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
ALL

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/nghttp3/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO alagoutte/nghttp3
```
Comment (JonLiu1993) [inline ports/nghttp3/portfile.cmake:3]: @alagoutte ,Thanks for your pr, we do not recommend adding private repositories, can you add ngtcp2/nghttp3?

File: ports/nghttp3/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "nghttp3",
+  "version": "0.7.0",
+  "description": "Implementation of RFC 9114 HTTP/3 mapping over QUIC and RFC 9204 QPACK in C",
+  "homepage": "https://github.com/nghttp2/nghttp2",
```
Comment (JonLiu1993) [inline ports/nghttp3/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/ngtcp2/nghttp3",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/nghttp3/portfile.cmake (hunk 1 above)
Comment (alagoutte) [reply ports/nghttp3/portfile.cmake:3]: Yes, i known but it is way i move to draft, the time of patch merged for fix ARM 32 bits

Comment (JonLiu1993) [reply ports/nghttp3/portfile.cmake:3]: @alagoutte, Thanks for your response, please convert pr to draft first

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
@alagoutte is this port intended to work if I installed nghttp2 in conjunction with nghttp3?

**Comment by JavierMatosD** [comment]
@alagoutte is this port intended to work if I have nghttp2 installed? It looks like it should be fine from my end, but I am just making sure. :)

**Comment by alagoutte** [comment]
> @alagoutte is this port intended to work if I have nghttp2 installed? It looks like it should be fine from my end, but I am just making sure. :)

Yes, it is 2 different tools (nghttp2 for HTTP/2 and HPACK, nghttp3 with ngtcp2 for QUIC with HTTP3 and QPACK)

**Comment by JavierMatosD** [comment]
Thanks!

**Pull request merged**